crates/ty_python_semantic/src/types/infer/builder.rs

```diff
@@ -4754,17 +4762,16 @@ impl<'db, 'ast> TypeInferenceBuilder<'db, 'ast> {
     }
 
     fn infer_return_statement(&mut self, ret: &ast::StmtReturn) {
-        if let Some(ty) =
-            self.infer_optional_expression(ret.value.as_deref(), TypeContext::default())
-        {
-            let range = ret
-                .value
-                .as_ref()
-                .map_or(ret.range(), |value| value.range());
-            self.record_return_type(ty, range);
-        } else {
-            self.record_return_type(Type::none(self.db()), ret.range());
-        }
+        self.infer_optional_expression(ret.value.as_deref(), TypeContext::default());
+        let range = ret
+            .value
+            .as_ref()
+            .map_or(ret.range(), |value| value.range());
+        let expression = ret
+            .value
+            .as_ref()
+            .map(|expr| ExpressionNodeKey::from(&**expr));
+        self.record_returnee(expression, range);
```

Comment:
Why do we need to store the expression key instead of storing the type directly?